OneCommunityGlobal/HGNRest
# Nikhil fix badges earned icon on profile page

# Description
![Task1](https://github.com/user-attachments/assets/b34964a0-f1ce-46dd-9fbc-469890391679)

## Related PRS (if any):
To test this backend PR you need to checkout the [#2691](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2691) frontend PR.
…

## Main changes explained:
-  Added a function in the badge controller to mark viewed badges.
-  Route added to badgerouter.

## How to test:
1. check into the current branch
2. do `npm install` and `...` to run this PR locally
3. follow the steps in the front end PR.
## Screenshots or videos of changes:

https://github.com/user-attachments/assets/d80085c8-ddaa-4212-bafb-0d1e97d88e4a



## Note:
Include the information the reviewers need to know.
